components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/streaming/websocket/WebSocketAnalyticsDataProvider.java

```diff
@@ -330,4 +336,24 @@ private String getApiContext() {
         return null;
     }
 
+    private long getRequestMediationLatency() {
+        if (WebSocketUtils.getPropertyFromChannel(Constants.BACKEND_START_TIME_PROPERTY, ctx) == null ||
+                (long) WebSocketUtils.getPropertyFromChannel(Constants.BACKEND_START_TIME_PROPERTY, ctx) == 0 &&
```

Comment:
Since we're relying on the `Constants.BACKEND_START_TIME_PROPERTY` property to detect whether the frame is from backend -> client or vice versa,
Shall we extract the following part into a method - named something like`isFrameFromBackend(ctx)`:
```java
if (WebSocketUtils.getPropertyFromChannel(Constants.BACKEND_START_TIME_PROPERTY, ctx) == null ||
                (long) WebSocketUtils.getPropertyFromChannel(Constants.BACKEND_START_TIME_PROPERTY, ctx) == 0)
```
This would make the code more readable.

```java
private boolean isFrameFromBackend(ChannelHandlerContext ctx) {
    return (WebSocketUtils.getPropertyFromChannel(Constants.BACKEND_START_TIME_PROPERTY, ctx) != null &&
                (long) WebSocketUtils.getPropertyFromChannel(Constants.BACKEND_START_TIME_PROPERTY, ctx) != 0);
}
```